src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,75 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
-    private static final Injector injector = new Injector();
+    private static final Injector INJECTOR = new Injector();
+    private static final Map<Class<?>, Class<?>> INTERFACE_IMPLS = new HashMap<>();
+    private static final Map<Class<?>, Object> INSTANCES = new HashMap<>();
 
     public static Injector getInjector() {
-        return injector;
+        return INJECTOR;
+    }
+
+    static {
+        INTERFACE_IMPLS.put(FileReaderService.class, FileReaderServiceImpl.class);
+        INTERFACE_IMPLS.put(ProductParser.class, ProductParserImpl.class);
+        INTERFACE_IMPLS.put(ProductService.class, ProductServiceImpl.class);
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("This class has no annotation Component");
+        }
```

Comment:
Can the interface have the annotation "Component"? I'd recommend you find the implementation first, and then - check the annotation.